src/sentry/api/endpoints/useravatar.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+
+from PIL import Image
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import UserAvatar, File
+from sentry.utils.compat import StringIO
+
+
+MIN_DIMENSION = 256
+
+MAX_DIMENSION = 1024
+
+
+class UserAvatarEndpoint(UserEndpoint):
+    FILE_TYPE = 'avatar.file'
+
+    def get(self, request, user):
+        return Response(serialize(user, request.user))
+
+    def is_valid_size(self, width, height):
+        if width != height:
+            return False
+        if width < MIN_DIMENSION:
+            return False
+        if width > MAX_DIMENSION:
+            return False
+        return True
+
+    def put(self, request, user):
+        if user != request.user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        photo_string = request.DATA.get('avatar_photo')
+        photo = None
+        if photo_string:
+            photo_string = photo_string.decode('base64')
+            with Image.open(StringIO(photo_string)) as img:
```

Comment:
it might be worth adding a small comment here about the important of this code since doing the size check before reading prevents a DoS vector